### PR TITLE
Feature design screen

### PR DESCRIPTION
**Pull request by thinhnguyen91**
Add ViewController

### CODE REVIEW

File: Weather/Controllers/Base/BaseNavigationController.swift [hunk 1]
```diff
@@ -13,7 +13,7 @@ class BaseNavigationController: UINavigationController {
     override func viewDidLoad() {
         super.viewDidLoad()
         self.navigationBar.tintColor = UIColor.whiteColor()
```
Comment (at-phuongvnc) [inline Weather/Controllers/Base/BaseNavigationController.swift]: please remove `self`

File: Weather/Controllers/HomeViewController/HomeViewController.swift [hunk 2]
```diff
@@ -13,11 +13,33 @@ class HomeViewController: BaseViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
 
+        self.alertGPS()
```
Comment (at-phuongvnc) [inline Weather/Controllers/HomeViewController/HomeViewController.swift]: please remove `self`

File: Weather/Controllers/HomeViewController/HomeViewController.swift [hunk 3]
```diff
@@ -13,11 +13,33 @@ class HomeViewController: BaseViewController {
...
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
 
     }
+    
+    func alertGPS() {
```
Comment (at-phuongvnc) [inline Weather/Controllers/HomeViewController/HomeViewController.swift]: You can write singleton manage GPS

File: Weather/Controllers/HomeViewController/HomeViewController.swift [hunk 4]
```diff
@@ -13,11 +13,33 @@ class HomeViewController: BaseViewController {
...
 
+        let alertController = UIAlertController(title: "Weather", message: "Do you want use GPS get your location.", preferredStyle: .Alert)
```
Comment (at-phuongvnc) [inline Weather/Controllers/HomeViewController/HomeViewController.swift]: please define title and message

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+import UIKit
+
+class LocationViewController: UIViewController {
+
+    @IBOutlet weak var tableView: UITableView!
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.tableView.delegate = self
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: please remove `self`

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    }
+}
+
+extension LocationViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+
+        return 50
+    }
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return 1
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: Don't need it in here. You can remove it

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        self.navigationController?.popToRootViewControllerAnimated(true)
+    }
+}
+
+extension LocationViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+
+        return 50
+    }
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: Don't need it in here. You can remove it

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+    @IBAction func backAction(sender: AnyObject) {
+        self.navigationController?.popToRootViewControllerAnimated(true)
+    }
+}
+
+extension LocationViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+
+        return 50
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: please define `height' of cell

File: Weather/Controllers/RegisterViewController/DetailViewController/DateViewController/DateViewController.swift [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+
+    }
+
+    @IBAction func backAction(sender: AnyObject) {
+        self.navigationController?.popViewControllerAnimated(true)
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/DetailViewController/DateViewController/DateViewController.swift]: remove `self`

File: Weather/Controllers/RegisterViewController/DetailViewController/DetailViewController.swift [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+  
+    }
+    
+    @IBAction func settingAction(sender: AnyObject) {
+        let mySettingVC = SettingViewController(nibName: "SettingViewController", bundle: nil)
+        self.navigationController?.pushViewController(mySettingVC, animated: true)
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/DetailViewController/DetailViewController.swift]: remove `self`

File: Weather/Controllers/RegisterViewController/DetailViewController/DetailViewController.swift [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    }
+    
+    @IBAction func backAction(sender: AnyObject) {
+        let myDateVC = DateViewController(nibName: "DateViewController", bundle: nil)
+        self.navigationController?.pushViewController(myDateVC, animated: true)
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/DetailViewController/DetailViewController.swift]: remove `self`

File: Weather/Controllers/RegisterViewController/DetailViewController/SettingViewController/SettingViewController.swift [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        // Do any additional setup after loading the view.
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+    @IBAction func backAction(sender: AnyObject) {
+        self.navigationController?.popViewControllerAnimated(true)
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/DetailViewController/SettingViewController/SettingViewController.swift]: remove `self`

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+        let nib = UINib(nibName: "CountryTableViewCell", bundle: nil)
+        tableView.registerNib(nib, forCellReuseIdentifier: "CountryCell")
+
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+
+    }
+
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: Please add `//MARK: - Action` in here

File: Weather/Controllers/HomeViewController/HomeViewController.swift [hunk 14]
```diff
@@ -13,11 +13,33 @@ class HomeViewController: BaseViewController {
...
+        }
+        alertController.addAction(noAction)
+
+        let yesAction = UIAlertAction(title: "Yes", style: .Default) { (action) in
+            let myRegisteeVC = RegisterViewController(nibName: "RegisterViewController", bundle: nil)
+            self.navigationController?.pushViewController(myRegisteeVC, animated: true)
+        }
+        alertController.addAction(yesAction)
+
+        self.presentViewController(alertController, animated: true) {
```
Comment (at-phuongvnc) [inline Weather/Controllers/HomeViewController/HomeViewController.swift]: please remove `self`

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    // MARK: - Action
+}
+
+extension LocationViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+
+        return kHeightCell
+    }
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: please remove it

File: Weather/Controllers/RegisterViewController/RegisterViewController.swift [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+     
+    }
+  
+    @IBAction func registerAction(sender: AnyObject) {
+        let myDetailVC = DetailViewController(nibName: "DetailViewController", bundle: nil)
+        self.navigationController?.pushViewController(myDetailVC, animated: true)
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/RegisterViewController.swift]: please remove `self`

**Comment by at-phuongvnc** [comment]
Please fix comment

File: Weather/Controllers/RegisterViewController/RegisterViewController.swift (hunk 16 above)
Comment (thinhnguyen91) [inline Weather/Controllers/RegisterViewController/RegisterViewController.swift]: screen shot 2016-07-04 at 14 33 49
when remove self, it error.

File: Weather/Controllers/HomeViewController/HomeViewController.swift (hunk 14 above)
Comment (thinhnguyen91) [inline Weather/Controllers/HomeViewController/HomeViewController.swift]: screen shot 2016-07-04 at 14 33 49
when remove self, it error.

File: Weather/Controllers/Base/BaseNavigationController.swift [hunk 17]
```diff
@@ -12,10 +12,11 @@ class BaseNavigationController: UINavigationController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        self.navigationBar.tintColor = UIColor.whiteColor()
-        self.navigationBar.barTintColor = UIColor.blueColor()
-        self.navigationBar.translucent = false
-        self.navigationBar.shadowImage = UIImage()
+        
+        navigationBar.tintColor = UIColor.whiteColor()
```
Comment (at-phuongvnc) [inline Weather/Controllers/Base/BaseNavigationController.swift]: You set it in AppDelegate.

File: Weather/Controllers/HomeViewController/HomeViewController.swift [hunk 18]
```diff
@@ -13,11 +13,33 @@ class HomeViewController: BaseViewController {
...
 
+        alertGPS()
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
 
     }
 
+    func alertGPS() {
```
Comment (at-phuongvnc) [inline Weather/Controllers/HomeViewController/HomeViewController.swift]: Please set private for method

File: Weather/Controllers/LocationViewController/CountryViewController/CountryCell/Define.swift [hunk 19]
```diff
@@ -0,0 +1,12 @@
...
+//  Weather
+//
+//  Created by ThinhNX on 7/4/16.
+//  Copyright © 2016 Intership. All rights reserved.
+//
+
+import Foundation
+
+let TITLE = "Weather"
+let MESSAGE = "Do you want use GPS get your location."
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryCell/Define.swift]: what is message ?

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 20]
```diff
@@ -0,0 +1,53 @@
+//
+//  LocationViewController.swift
+//  Weather
+//
+//  Created by ThinhNX on 7/1/16.
+//  Copyright © 2016 Intership. All rights reserved.
+//
+
+import UIKit
+let kHeightCell = CGFloat(50)
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: Add it in `struct Options`

File: Weather/Controllers/RegisterViewController/DetailViewController/DateViewController/DateViewController.swift [hunk 21]
```diff
@@ -0,0 +1,27 @@
...
+        super.viewDidLoad()
+
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+
+    }
+
+    @IBAction func backAction(sender: AnyObject) {
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/DetailViewController/DateViewController/DateViewController.swift]: please set private of func

File: Weather/Controllers/RegisterViewController/DetailViewController/DetailViewController.swift [hunk 22]
```diff
@@ -0,0 +1,33 @@
...
+        super.viewDidLoad()
+
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+
+    }
+
+    @IBAction func settingAction(sender: AnyObject) {
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/DetailViewController/DetailViewController.swift]: missing `private`

File: Weather/Controllers/RegisterViewController/DetailViewController/DetailViewController.swift [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+        let mySettingVC = SettingViewController(nibName: "SettingViewController", bundle: nil)
+        navigationController?.pushViewController(mySettingVC, animated: true)
+    }
+
+    @IBAction func backAction(sender: AnyObject) {
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/DetailViewController/DetailViewController.swift]: missing `private`

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+class LocationViewController: UIViewController {
+
+    @IBOutlet weak var tableView: UITableView!
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: missing `private

File: Weather/Controllers/LocationViewController/CountryViewController/CountryCell/Define.swift [hunk 25]
```diff
@@ -0,0 +1,12 @@
...
+//  Define.swift
+//  Weather
+//
+//  Created by ThinhNX on 7/4/16.
+//  Copyright © 2016 Intership. All rights reserved.
+//
+
+import Foundation
+
+let TITLE = "Weather"
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryCell/Define.swift]: Add it in class and use static

File: Weather/Controllers/RegisterViewController/DetailViewController/SettingViewController/SettingViewController.swift [hunk 26]
```diff
@@ -0,0 +1,29 @@
...
+
+        // Do any additional setup after loading the view.
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+    @IBAction func backAction(sender: AnyObject) {
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/DetailViewController/SettingViewController/SettingViewController.swift]: missing `private`

File: Weather/Controllers/RegisterViewController/DetailViewController/SettingViewController/SettingViewController.swift [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+//  SettingViewController.swift
+//  Weather
+//
+//  Created by ThinhNX on 7/1/16.
+//  Copyright © 2016 Intership. All rights reserved.
+//
+
+import UIKit
+
+class SettingViewController: UIViewController {
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/DetailViewController/SettingViewController/SettingViewController.swift]: please use `BaseViewController`

File: Weather/Controllers/LocationViewController/CountryViewController/CountryCell/CountryTableViewCell.swift [hunk 28]
```diff
@@ -0,0 +1,24 @@
...
+//  CountryTableViewCell.swift
+//  Weather
+//
+//  Created by ThinhNX on 7/4/16.
+//  Copyright © 2016 Intership. All rights reserved.
+//
+
+import UIKit
+
+class CountryTableViewCell: UITableViewCell {
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryCell/CountryTableViewCell.swift]: please use `BaseTableViewCell`

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 29]
```diff
@@ -0,0 +1,53 @@
...
+//  LocationViewController.swift
+//  Weather
+//
+//  Created by ThinhNX on 7/1/16.
+//  Copyright © 2016 Intership. All rights reserved.
+//
+
+import UIKit
+let kHeightCell = CGFloat(50)
+class LocationViewController: UIViewController {
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: please use `BaseViewController`

File: Weather/Controllers/RegisterViewController/DetailViewController/DateViewController/DateViewController.swift [hunk 30]
```diff
@@ -0,0 +1,27 @@
...
+//  DateViewController.swift
+//  Weather
+//
+//  Created by ThinhNX on 7/4/16.
+//  Copyright © 2016 Intership. All rights reserved.
+//
+
+import UIKit
+
+class DateViewController: UIViewController {
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/DetailViewController/DateViewController/DateViewController.swift]: please use `BaseViewController`

File: Weather/Controllers/RegisterViewController/DetailViewController/DetailViewController.swift [hunk 31]
```diff
@@ -0,0 +1,33 @@
...
+//  DetailViewController.swift
+//  Weather
+//
+//  Created by ThinhNX on 7/1/16.
+//  Copyright © 2016 Intership. All rights reserved.
+//
+
+import UIKit
+
+class DetailViewController: UIViewController {
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/DetailViewController/DetailViewController.swift]: please use `BaseViewController`

File: Weather/Controllers/RegisterViewController/RegisterViewController.swift [hunk 32]
```diff
@@ -0,0 +1,27 @@
...
+//  RegisterViewController.swift
+//  Weather
+//
+//  Created by ThinhNX on 7/1/16.
+//  Copyright © 2016 Intership. All rights reserved.
+//
+
+import UIKit
+
+class RegisterViewController: UIViewController {
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/RegisterViewController.swift]: please use `BaseViewController`

File: Weather/Controllers/LocationViewController/CountryViewController/CountryCell/Define.swift [hunk 33]
```diff
@@ -7,6 +7,7 @@
 //
 
 import Foundation
-
-let TITLE = "Weather"
-let MESSAGE = "Do you want use GPS get your location."
\ No newline at end of file
+struct Alert {
+    static let TITLE = "Weather"
+    static let MESSAGE = "Do you want use GPS get your location."
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryCell/Define.swift]: please rename message
EX: MessageGPS

**Comment by at-phuongvnc** [comment]
LGTM 👍

**Pull request merged**